Emby.Dlna/RSSDP/SsdpServerPublisher.cs

```diff
@@ -0,0 +1,642 @@
+                    ["LOCATION"] = rootDevice.Location.ToString(),
+                    ["SERVER"] = _networkManager.SsdpServer,
+                    ["NTS"] = "ssdp:alive",
+                    ["NT"] = notificationType,
+                    ["USN"] = uniqueServiceName
+                };
+
+                var message = BuildMessage("NOTIFY * HTTP / 1.1", values);
+                _logger.LogDebug("-> SSDP:ALIVE : {1} - {2}", multicastAddresses[a], uniqueServiceName);
+                tasks[a] = _socketServer.SendMulticastMessageAsync(message, rootDevice.Address);
+            }
+
+            return Task.WhenAll(tasks);
+        }
+
+        private Task SendByeByeNotifications(SsdpDevice device, bool isRoot)
+        {
+            var tasks = new List<Task>();
+            if (isRoot)
+            {
+                tasks.Add(SendByeByeNotification(device, UpnpRootDevice, GetUsn(device.Udn, UpnpRootDevice)));
+                if (this.SupportPnpRootDevice)
+                {
+                    tasks.Add(SendByeByeNotification(device, "pnp:rootdevice", GetUsn(device.Udn, "pnp:rootdevice")));
+                }
+            }
+
+            tasks.Add(SendByeByeNotification(device, device.Udn, device.Udn));
+            tasks.Add(SendByeByeNotification(
+                device,
+                string.Format(CultureInfo.InvariantCulture, "urn:{0}", device.FullDeviceType),
+                GetUsn(device.Udn, device.FullDeviceType)));
+
+            foreach (var childDevice in device.Devices)
+            {
+                tasks.Add(SendByeByeNotifications(childDevice, false));
+            }
+
+            return Task.WhenAll(tasks);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "byebye", Justification = "Correct value for this type of notification in SSDP.")]
+        private Task SendByeByeNotification(SsdpDevice device, string notificationType, string uniqueServiceName)
+        {
+            string[] multicastAddresses = { "239.255.255.250:1900", "[ff02::C]:1900", "[ff05::C]:1900" };
+            Task[] tasks = { Task.CompletedTask, Task.CompletedTask, Task.CompletedTask };
+            int count = _networkManager.IsIP6Enabled ? 2 : 0;
+
+            for (int a = 0; a <= count; a++)
+            {
+                var values = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+                {
+                    ["HOST"] = multicastAddresses[a],
+                    ["DATE"] = DateTime.UtcNow.ToString("r", CultureInfo.InvariantCulture),
+                    ["SERVER"] = _networkManager.SsdpServer,
+                    ["NTS"] = "ssdp:byebye",
+                    ["NT"] = notificationType,
+                    ["USN"] = uniqueServiceName
+                };
+
+                var message = BuildMessage("NOTIFY* HTTP/ 1.1", values);
```

Comment:
I think this might need to be `NOTIFY * HTTP/1.1`

Reply:
Thanks - found and corrected it. That will be the annoying VS2019 format on paste option. Just disabled it.

Reply:
> 
> 
> Thanks - found and corrected it. That will be the annoying VS2019 format on paste option. Just disabled it.

The spec shows  a space before and after the `*` (as do the successful messages I pull from wireshark/device sniffer) - not sure if it matters or not but should probably be consistent...testing these changes now...

Reply:
Corrected. Thanks